### PR TITLE
Updated static deploy to also record majorVersion-latest.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**

### CODE REVIEW

File: lib/deploy-static.js [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
+  // Only update major version pointers if it's a valid version and not a pre-release.
+  if (semver.valid(settings.version) && !semver.prerelease(settings.version)) {
+    const versionMajor = semver.major(settings.version).toString();
+    await register(settings, versionMajor, versionMajor);
+    await register(settings, settings.version, `${versionMajor}-latest`)
+  } else {
+    logger.info('Not updating latest major release as version in package.json is invalid or is a pre-release.');
+  }
+
+  return Promise.resolve();
```
Comment (Blackbaud-SteveBrush) [inline lib/deploy-static.js]: I believe this return statement is redundant and can be removed.

File: lib/deploy-static.js [hunk 2]
```diff
@@ -1,18 +1,45 @@
...
+async function deploy(settings) {
+
+  // Register exact version
+  await register(settings, settings.version, settings.version);
+
+  // Only update major version pointers if it's a valid version and not a pre-release.
+  if (semver.valid(settings.version) && !semver.prerelease(settings.version)) {
+    const versionMajor = semver.major(settings.version).toString();
+    await register(settings, versionMajor, versionMajor);
+    await register(settings, settings.version, `${versionMajor}-latest`)
```
Comment (Blackbaud-SteveBrush) [inline lib/deploy-static.js]: Needs a semicolon.

File: lib/deploy-static.js [hunk 3]
```diff
@@ -1,18 +1,45 @@
...
 }
 
+async function deploy(settings) {
+
+  // Register exact version
+  await register(settings, settings.version, settings.version);
+
+  // Only update major version pointers if it's a valid version and not a pre-release.
+  if (semver.valid(settings.version) && !semver.prerelease(settings.version)) {
+    const versionMajor = semver.major(settings.version).toString();
```
Comment (Blackbaud-SteveBrush) [inline lib/deploy-static.js]: This is pedantic, but could we rename to `majorVersion`?

File: README.md [hunk 4]
```diff
@@ -27,5 +27,6 @@ All of the following options are required and overridable via the CLI.  For exam
 | `azureStorageAccessKey` | `null`  |
 | `azureStorageTableName` | "spa"   |
 | `isStaticClient`        | `false` |
+| `updateMajorVersion`    | `false` |
```
Comment (Blackbaud-SteveBrush) [inline README.md]: From our conversation, I believe this needs to be removed.

File: lib/deploy-static.js (hunk 3 above)
Comment (Blackbaud-BobbyEarl) [reply lib/deploy-static.js]: As per our discussion, keeping this as is for consistency with other variable names.

**Comment by Blackbaud-BobbyEarl** [comment]
Addressed your feedback, thanks @Blackbaud-SteveBrush!

**Pull request merged**